Sources/PortalSwift/Core/Mpc/PortalMpc.swift

```diff
@@ -499,19 +476,19 @@ public class PortalMpc {
             let decryptionKey = try await storage.read()
 
             usingProgressCallback?(MpcStatus(status: .decryptingShare, done: false))
-            let decryptedString = try await storage.decrypt(cipherText, withKey: decryptionKey)
-            guard let decryptedData = decryptedString.data(using: .utf8) else {
-              throw MpcError.unexpectedErrorOnRecover("Unable to parse decrypted data.")
-            }
+            let decryptedSharesString = try await storage.decrypt(cipherText, withKey: decryptionKey)
             usingProgressCallback?(MpcStatus(status: .parsingShare, done: false))
-            let backupResponse = try await parseRecoveryInput(data: decryptedData)
+            let shares = try await formatShares(sharesJSONString: decryptedSharesString)
 
             usingProgressCallback?(MpcStatus(status: .generatingShare, done: false))
             var recoverResponse: PortalMpcGenerateResponse = [:]
             if let shares = try? await keychain.getShares() {
               recoverResponse = shares
             }
-            if let ed25519Share = backupResponse[PortalCurve.ED25519.rawValue] {
+
+            var recoveredShareCount = 0
```

Comment:
what is this counter for?

Reply:
i see its used to check fi we were able to recover any shares but i think we can check first if the data object has either of the keys before we even run it. this is likely a nit but just calling out its a bit odd to me

Reply:
Yupp exactly that was its intent, but agreed it could be simplified 💯 Going it to leave it as is for now since I don't know if it's worth rerunning all the tests again for, but lmk if you feel otherwise!